### PR TITLE
changing docker pull to proxy cache

### PR DESCRIPTION
**Pull request by Akyiaa**
- Update to Base Dockerfile so it uses Proxy and simple Pipeline to build and push Base image
- replaced docker.io default registry with new proxy cache
replaced docker.io with harbor.galasa.dev/docker_proxy_cache. added the proxy cache in front of images for dockerfiles

### CODE REVIEW

File: dockerfiles/cli/cli-dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.18.5
+FROM harbor.galasa.dev/docker_proxy_cache/alpine:3.18.5
```
Comment (jadecarino) [inline dockerfiles/cli/cli-dockerfile:1]: I believe this will need 'library' so harbor.galasa.dev/docker_proxy_cache/library/alpine:3.18.5 - same for the rest of the alpine ones if you could make that change please :)

**Review by jadecarino — CHANGES_REQUESTED** [review]

File: build-images/github-webhook-receiver/go.mod [hunk 2]
```diff
@@ -12,5 +12,6 @@ require (
 	github.com/kr/pretty v0.3.1 // indirect
 	github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2 // indirect
 	github.com/rogpeppe/go-internal v1.10.0 // indirect
+	github.com/spf13/viper v1.18.2 // indirect
```
Comment (jadecarino) [inline build-images/github-webhook-receiver/go.mod:15]: Can we remove this from this PR please if the change is not related to this PR?

**Review by jadecarino — CHANGES_REQUESTED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful
